ci/infra/vmware/cloud-init/common.tpl

```diff
@@ -56,6 +56,7 @@ ${commands}
 
 bootcmd:
   # Hostnames from DHCP - otherwise `localhost` will be used
-  - /usr/bin/sed -ie "s#DHCLIENT_SET_HOSTNAME=\"no\"#DHCLIENT_SET_HOSTNAME=\"yes\"#" /etc/sysconfig/network/dhcp
+  - /usr/bin/sed -ie "s#DHCLIENT_SET_HOSTNAME=\"yes\"#DHCLIENT_SET_HOSTNAME=\"no\"#" /etc/sysconfig/network/dhcp
+  - /usr/bin/hostnamectl set-hostname ${hostname}
```

Comment:
Does this need to set every time node reboot? If not, we could put it in `runcmd` on first boot?  Also, does `hostname:` for cloud-init work?

Reply:
It also needs to change `preserve_hostname` from _false_ to _true_ in `/etc/cloud/cloud.cfg` and the node reboot is needed (otherwise it would be changed after skuba node bootstrap/join :thinking: ).

Two challenges from now:
1. In runcmd stage, change `preserve_hostname` from _false_ to _true_ in `/etc/cloud/cloud.cfg` failed, because the file `/etc/cloud/cloud.cfg` is still not exist.
2. I'm still testing how to achieve this _without_ the node reboot.

Reply:
All done! After tracing cloud-init code and journal log, I found the _key_ to make it happen:
1. Set hostname by `hostname` config key.
2. `netconfig update -f` doesn't work, but instead `systemctl restart wicked` works to let wicked read new configuration file `/etc/sysconfig/network/dhcp`.